[musl] Unsupported parameter type 'snd_htimestamp_t *' (aka 'timespec *')

Build thunks using musl fails with:

```
[45/71] Linking CXX shared library HostLibs_64/libdrm-host.so
ninja: job failed: cd /run/host/home/pepper/dev/ghrepos/FEX/build_tests/ThunkLibs/HostLibs && /run/host/home/pepper/dev/ghrepos/FEX/build_tests/Bin/thunkgen /run/host/home/pepper/dev/ghrepos/FEX/ThunkLibs/HostLibs/../libasound/libasound_interface.cpp libasound -host /run/host/home/pepper/dev/ghrepos/FEX/build_tests/ThunkLibs/HostLibs/gen_64/thunkgen_host_libasound.inl /nix/store/2wzi1fs6j0g0199l8jagnranwpaiaylw-fex-dev-rootfs -- -std=c++20 -DARCHITECTURE_arm64=1 -isystem/run/host/home/pepper/dev/ghrepos/FEX/ThunkLibs/HostLibs/../include
In file included from /run/host/home/pepper/dev/ghrepos/FEX/ThunkLibs/HostLibs/../libasound/libasound_interface.cpp:3:
In file included from /usr/include/alsa/asoundlib.h:58:
/usr/include/alsa/pcm.h:566:5: error: Unsupported parameter type 'snd_htimestamp_t *' (aka 'timespec *')
  566 | int snd_pcm_htimestamp(snd_pcm_t *pcm, snd_pcm_uframes_t *avail, snd_htimestamp_t *tstamp);
      |     ^
1 error generated.
```

--- a/ThunkLibs/libasound/libasound_interface.cpp
+++ b/ThunkLibs/libasound/libasound_interface.cpp
@@ -147,6 +147,10 @@ struct fex_gen_type<snd_timer_t> : fexgen::opaque_type {};
 template<>
 struct fex_gen_type<snd_timestamp_t> : fexgen::opaque_type {};
 
+// has empty padding members on musl
+template<>
+struct fex_gen_type<snd_htimestamp_t> : fexgen::assume_compatible_data_layout {};
+
 template<>
 struct fex_gen_type<FILE> : fexgen::opaque_type {};
 

